src/treebuilders/add.cpp

```diff
@@ -126,67 +128,143 @@ template <int D> void add(double prec, FunctionTree<D> &out, FunctionTreeVector<
     print::separator(10, ' ');
 }
 
-template <int D> void add(double prec, FunctionTree<D> &out, std::vector<FunctionTree<D> *> &inp, int maxIter, bool absPrec) {
-    FunctionTreeVector<D> inp_vec;
+template <int D, typename T> void add(double prec, FunctionTree<D, T> &out, std::vector<FunctionTree<D, T> *> &inp, int maxIter, bool absPrec, bool conjugate) {
+    FunctionTreeVector<D, T> inp_vec;
     for (auto &t : inp) inp_vec.push_back({1.0, t});
-    add(prec, out, inp_vec, maxIter, absPrec);
+    add(prec, out, inp_vec, maxIter, absPrec, conjugate);
 }
 
-template void add<1>(double prec,
-                     FunctionTree<1> &out,
+template void add<1, double>(double prec,
```

Comment:
yes, a macro would go a very long way to reduce the boilerplate also for these instantiations!